torch/csrc/jit/tensorexpr/kernel.cpp

```diff
@@ -273,7 +273,29 @@ std::vector<ExprHandle> TensorExprKernel::inferSizesForValue(
       return shape;
     }
 
-    case aten::cat:
+    case aten::cat: {
+      // In JIT IR, aten::cat usually appears with the following nodes around
+      // it:
+      //   %dim : int = prim::Constant[value=0]()
+      //   %inputs : Tensor[] = prim::ListConstruct(%a, %b, ...)
+      //   %cat_output : Tensor = aten::cat(%inputs, %dim)
+      // Shapes of the input tensors could only differ at the dimension %dim.
+      // The sizes of the output tensor on that dimension is a sum of the
+      // corresponding sizes of the input tensors, the other dimension have the
+      // same sizes.
+      auto const& n = v->node();
+      auto inputs = n->inputs()[0]->node()->inputs();
+      int64_t dim = n->inputs()[1]->node()->i(attr::value);
```

Comment:
Does this assume the %dim is an integer constant?  What happens if it isn't?

Reply:
Then we will fail miserably :) I will add some checks for this!

Reply:
To expand a bit: when we add `aten::cat` nodes to a fusion group we do check that this is true: https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/passes/tensorexpr_fuser.cpp#L235-L243

I added an assert so that we fail loudly if something goes not like we expected.